src/utils/_getOptionValue.js

```diff
@@ -0,0 +1,8 @@
+// Composes getOption with the intent of
+// evaluating a target's property or option.
+// Useful when class property like `events` or `regions`
+// could either be an object or function.
+export default function(target, optionName, params) {
```

Comment:
yea, honestly, not feeling it... I think `this.getValue(this.getOption('foo'))` is fine in most cases